source/common/http/conn_pool_base.h

```diff
@@ -151,7 +152,9 @@ class FixedHttpConnPoolImpl : public HttpConnPoolImplBase {
                         CreateCodecFn codec_fn, std::vector<Http::Protocol> protocol)
       : HttpConnPoolImplBase(host, priority, dispatcher, options, transport_socket_options,
                              random_generator, state, protocol),
-        codec_fn_(codec_fn), client_fn_(client_fn) {}
+        codec_fn_(codec_fn), client_fn_(client_fn), protocol_(protocol[0]) {
+    ASSERT(protocol.size() == 1);
```

Comment:
nit: I'm realizing that this predates this change, but should this parameter be called `protocols` since it's a vector?